From b6cdb9a9f5eb1c4ae5b7769d90a79f29853a0fe2 Mon Sep 17 00:00:00 2001
From: kito <kito@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 7 Aug 2019 03:12:34 +0000
Subject: [PATCH 10/14] RISC-V: Handle g extension in multilib-generator

gcc/ChangeLog

	* gcc/config/riscv/multilib-generator: (canonical_order): Add 'g'.
	(arch_canonicalize): Support rv32g and rv64g and fix error
	handling.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@274156 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                       | 6 ++++++
 gcc/config/riscv/multilib-generator | 6 +++---
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index c97098bb451..3734e9a46ca 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,9 @@
+2019-08-07  Kito Cheng  <kito.cheng@sifive.com>
+
+	* gcc/config/riscv/multilib-generator: (canonical_order): Add 'g'.
+	(arch_canonicalize): Support rv32g and rv64g and fix error
+	handling.
+
 2019-08-06  Kito Cheng  <kito.cheng@sifive.com>
 
 	* gcc/config/riscv/multilib-generator: (canonical_order): New.
diff --git a/gcc/config/riscv/multilib-generator b/gcc/config/riscv/multilib-generator
index e58231c2756..a4125ff01fd 100755
--- a/gcc/config/riscv/multilib-generator
+++ b/gcc/config/riscv/multilib-generator
@@ -36,17 +36,17 @@ abis = collections.OrderedDict()
 required = []
 reuse = []
 
-canonical_order = "mafdqlcbjtpvn"
+canonical_order = "mafdgqlcbjtpvn"
 
 def arch_canonicalize(arch):
   # TODO: Support Z, S, H, or X extensions.
   # TODO: Support implied extensions, e.g. D implied F in latest spec.
   # TODO: Support extension version.
   new_arch = ""
-  if arch[:5] in ['rv32e', 'rv32i', 'rv64i']:
+  if arch[:5] in ['rv32e', 'rv32i', 'rv32g', 'rv64i', 'rv64g']:
     new_arch = arch[:5]
   else:
-    raise Exception("Unexpected arch: `%d`" % arch[:5])
+    raise Exception("Unexpected arch: `%s`" % arch[:5])
 
   # Find any Z, S, H or X
   long_ext_prefixes = ['z', 's', 'h', 'x']
-- 
2.20.1

